mne_bids/read.py

```diff
@@ -74,6 +75,48 @@ def _read_raw(raw_fpath, electrode=None, hsp=None, hpi=None, config=None,
     return raw
 
 
+def _handle_info_reading(sidecar_fname, raw):
+    """Read associated sidecar.json and populate raw.
+
+    Handle PowerLineFrequency of recording.
+    """
+    with open(sidecar_fname, "r") as fin:
+        sidecar_json = json.load(fin)
+
+    line_freq = sidecar_json.get("PowerLineFrequency")
+    if line_freq is None and raw.info["line_freq"] is None:
+        # defaults to 50 Hz if neither raw, or sidecar json
+        # has Power Line Freq set
+        raw.info["line_freq"] = 50
```

Comment:
This is dangerous -- why default to 50 Hz?

Reply:
Set it to `None` if neither is available? The user can always edit the `raw.info['line_freq']`

Reply:
Inside, write.py, `_sidecar_json`, there is this line:

```
    powerlinefrequency = raw.info.get('line_freq', None)
    if powerlinefrequency is None:
        warn('No line frequency found, defaulting to 50 Hz')
        powerlinefrequency = 50
```

I was following this convention. Does it make sense to set it default as 50 on write, but not on read?

Reply:
humm ... that's really old code, feel free to fix it in this PR

Reply:
I don't exactly remember but it could be that this is a required param for the validator.

Maybe you can do better by estimating the power line frequency (50 or 60 Hz) by just looking at the peak of the PSD if it's `None`. and not filtered.

Reply:
Yeah looks like it is a "REQUIRED" section, so it makes sense to at least write something, but I suppose then the PSD peak estimation could be done for both `write.py` and `read.py`

https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/04-intracranial-electroencephalography.html

I'll take a stab later.

Reply:
you could always write it as `'n/a'` but that's the lazy approach. It would be great if you can try the PSD approach and see if it works